src/Microsoft.AspNetCore.Authentication/AuthenticationHandler.cs

```diff
@@ -233,15 +233,40 @@ public bool ShouldHandleScheme(string authenticationScheme, bool handleAutomatic
             }
         }
 
+        /// <summary>
+        /// Handle the authentication for once.
```

Comment:
Nit: slightly better wording might be:

Used to ensure HandleAuthenticateAsync is only invoked once, subsequent calls will return the same result.